py/sentry_streams/flink/flink_adapter.py

```diff
@@ -25,10 +26,33 @@ class FlinkAdapter(StreamAdapter):
     # that send data to a next step that
     # performs serialization (e.g. Map --> Sink)
 
-    def __init__(self, config: MutableMapping[str, Any], env: StreamExecutionEnvironment) -> None:
+    def __init__(self, config: PipelineConfig, env: StreamExecutionEnvironment) -> None:
         self.environment_config = config
         self.env = env
 
+    @classmethod
+    def build(cls, config: PipelineConfig) -> Self:
+        env = StreamExecutionEnvironment.get_execution_environment()
+
+        flink_config = config.get("flink", {})
```

Comment:
Is the idea that, at least for now, each streaming pipeline will have a large config dict that stores configuration for each different runtime? 

Do we eventually want to move to defining general configs which can be "translated" into any specific runtime's config?

Reply:
> Is the idea that, at least for now, each streaming pipeline will have a large config dict that stores configuration for each different runtime?

At this point yes. It may be structured though. 
as we iterate many settings could become generic. So yes to your second question as well.